File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/MediaVariantV1.java
package com.cleverferret.v2.core.database.schema;

import com.cleverferret.v2.core.common.VersionedContract;

/**
 * Canonical format/storage variant row that hangs off a MediaItem identity.
 */
public record MediaVariantV1(
        String mediaVariantId,
        String mediaItemId,
        String sourceType,
        String uri,
        String fileName,
        String mimeType,
        String format,
        Long byteSize,
        String contentHash,
        String drmType,
        boolean downloadable,
        boolean availableOffline,
        long discoveredAtEpochMs,
        long scannedAtEpochMs
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/core/database/src/main/java/com/cleverferret/v2/core/database/schema/MediaVariantV1.kt
package com.cleverferret.v2.core.database.schema

import com.cleverferret.v2.core.common.VersionedContract

data class MediaVariantV1(
    val mediaVariantId: String,
    val mediaItemId: String,
    val sourceType: String,
    val uri: String,
    val fileName: String,
    val mimeType: String,
    val format: String,
    val byteSize: Long?,
    val contentHash: String,
    val drmType: String,
    val downloadable: Boolean,
    val availableOffline: Boolean,
    val discoveredAtEpochMs: Long,
    val scannedAtEpochMs: Long
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
